openedx/core/djangoapps/user_api/accounts/views.py

```diff
@@ -683,6 +683,52 @@ def retirement_queue(self, request):
         except RetirementStateError as exc:
             return Response(text_type(exc), status=status.HTTP_400_BAD_REQUEST)
 
+    def retirements_by_status_and_date(self, request):
+        """
+        GET /api/user/v1/accounts/retirements_by_status_and_date/
+        ?start_date=2018-09-05&end_date=2018-09-07&state=COMPLETE
+
+        Returns a list of UserRetirementStatusSerializer serialized
+        RetirementStatus rows in the given state that were created in the
+        retirement queue between the dates given. Date range is inclusive,
+        so to get one day you would set both dates to that day.
+        """
+        try:
+            start_date = datetime.datetime.strptime(request.GET['start_date'], '%Y-%m-%d')
+            end_date = datetime.datetime.strptime(request.GET['end_date'], '%Y-%m-%d')
+            now = datetime.datetime.now()
+            if start_date > now or end_date > now or start_date > end_date:
+                raise RetirementStateError('Dates must be today or earlier, and start must be earlier than end.')
+
+            # Add a day to make sure we get all the way to 23:59:59.999, this is compared "lt" in the query
+            # not "lte".
+            end_date += datetime.timedelta(days=1)
+            state = request.GET['state']
+
+            if not state:
+                raise RetirementStateError('Param "state" is required.')
```

Comment:
Is this check necessary since a KeyError will be raised if `state` isn't present?
https://docs.djangoproject.com/en/2.0/ref/request-response/#django.http.QueryDict.__getitem__

Reply:
Nope, will fix.